### PR TITLE
SPL aggregate funcs, eval informational, time and conversion functions

### PR DESCRIPTION
**Pull request by venkateshamatam**
This PR includes docs for - 

1. All supported stats sub commands
2. Timechart sub commands
3. Eval subcommands - Informational functions
4. Eval subcommands - Date and time functions
5. Eval subcommands - Conversion Functions

### CODE REVIEW

File: docs/spl-docs/aggregate-functions/agg-functions.md
Comment (AndrewHess) [inline docs/spl-docs/aggregate-functions/agg-functions.md:138]: nit: summing latency is probably not useful; the average latency would probably be more useful for making decisions.

Maybe summing something else would be more useful. For example, summing revenue from subscriptions per month per city

File: docs/spl-docs/evaluation-functions/informational-functions.md
Comment (venkateshamatam) [reply docs/spl-docs/evaluation-functions/informational-functions.md]: I have updated the doc to specify what the valuec an be, possible outputs, and a use-case example with Problem, solution and the implementation

File: docs/spl-docs/evaluation-functions/informational-functions.md
Comment (venkateshamatam) [reply docs/spl-docs/evaluation-functions/informational-functions.md:1]: I have added use-case examples for all functions in this file in the same format as it is in other parts of the doc

File: docs/spl-docs/evaluation-functions/conversion-functions.md
Comment (venkateshamatam) [reply docs/spl-docs/evaluation-functions/conversion-functions.md]: I have clarified the above things in the latest commit, added valid and invalid fields for Value and cases where value is null, also added Formatting details to clarify what `hex`, `commas` and `duration` will do.

File: docs/spl-docs/evaluation-functions/time-functions.md
Comment (venkateshamatam) [reply docs/spl-docs/evaluation-functions/time-functions.md]: I have specified the unit and timezone in the latest commit.

File: docs/spl-docs/evaluation-functions/time-functions.md
Comment (venkateshamatam) [reply docs/spl-docs/evaluation-functions/time-functions.md]: I have specified the invalid inputs case and added use-case examples, for specifying the valid specifiers `format` can contain, there are multiple specifiers in the `formatTime` method (linked below) - how should we list these? Should we have a table with all the valid specifiers? Should the table contain examples, or written description for each or both? 
https://github.com/siglens/siglens/blob/726bd106f7a40a9e2c9d44def394faecd9837a92/pkg/segment/structs/evaluationstructs.go#L1624-L1681

File: docs/spl-docs/evaluation-functions/time-functions.md
Comment (venkateshamatam) [reply docs/spl-docs/evaluation-functions/time-functions.md]: I have addressed this in the latest commit

File: docs/spl-docs/aggregate-functions/agg-functions.md
Comment (venkateshamatam) [reply docs/spl-docs/aggregate-functions/agg-functions.md:67]: I have added use-case examples for all stats/timechart commands

File: docs/spl-docs/evaluation-functions/conversion-functions.md
Comment (AndrewHess) [inline docs/spl-docs/evaluation-functions/conversion-functions.md:135]: nit: are you saying they'd use `tostring` to format it and show it to the user? It would generally only be employees who are using Splunk from logs that were generated from their program running, so I don't think showing Splunk results to a user would make sense

File: docs/spl-docs/evaluation-functions/informational-functions.md
Comment (AndrewHess) [inline docs/spl-docs/evaluation-functions/informational-functions.md:52]: > This is especially useful in conditional expressions and for filtering records based on numeric field values.

I think that's not quite correct, because `isnum` returns a boolean, so you can't use it to filter records based on numeric field values. Unless you mean filtering records that have any (rather than a specific) numeric field value

File: docs/spl-docs/evaluation-functions/informational-functions.md
Comment (AndrewHess) [inline docs/spl-docs/evaluation-functions/informational-functions.md:180]: Since a log line won't be changed once it's been ingested, I think this example doesn't quite make sense. Because if you first ingested a log line that this query would mark as `Awaiting`, and then the feedback was submitted so you got another log line ingested that this query would mark as `Received`, you'd still have one log line that gets marked as `Awaiting`. Maybe changing `Awaiting` to something else, or using a different example would make this more relevant.

You might be over thinking this. Something like `validate(isnotnull(<some-field>, ...))` could be a good enough example

Similar comment for your `isnull` Use-Case Example

File: docs/spl-docs/evaluation-functions/informational-functions.md
Comment (AndrewHess) [inline docs/spl-docs/evaluation-functions/informational-functions.md:254]: You're just repeating your Example 2 (lines 232-237); you don't need this duplication. Either give a different example here or remove Example 2.

Either way, I think a little more detailed example here could be good. For example, maybe this field is supposed to be a number but if we get a string we'll try converting it (e.g., strip units, convert from hex, etc.), but booleans and NULLs are invalid. Or if you have a better idea, go with that. But the purpose of this is to make it easy for people to read an example and then be able to apply it to their own use case to solve their problem; by giving an example with steps after using `typeof`, we can make this easier for them

File: docs/spl-docs/evaluation-functions/time-functions.md
Comment (AndrewHess) [inline docs/spl-docs/evaluation-functions/time-functions.md:45]: Do we support the `table` command yet?

File: docs/spl-docs/evaluation-functions/informational-functions.md
Comment (venkateshamatam) [reply docs/spl-docs/evaluation-functions/informational-functions.md:52]: yes, this actually means filtering records that have any numeric field value, I'll update the statement to make it more clear.

File: docs/spl-docs/evaluation-functions/time-functions.md
Comment (AndrewHess) [reply docs/spl-docs/evaluation-functions/time-functions.md]: If there's already stable, correct documentation for these, we can link to that; otherwise, we should write our own. A table sounds like a good idea to me. We want to communicate to the reader accurately and concisely. I would guess that would mean a written description and possible examples for some of them, but use your judgement

**Review by AndrewHess — COMMENTED** [review]
Can you also change this PR so it will merge into `develop` instead of merging into `spl-docs`? We squashed `spl-docs` onto `develop`, so now if we update `spl-docs` with this PR, I think we'll end up with merge conflicts when we try to get this into `develop`

File: docs/spl-docs/aggregate-functions/agg-functions.md
Comment (AndrewHess) [inline docs/spl-docs/aggregate-functions/agg-functions.md:151]: typo: on line 151, `list` should be `lists`

File: docs/spl-docs/evaluation-functions/informational-functions.md
Comment (AndrewHess) [inline docs/spl-docs/evaluation-functions/informational-functions.md:89]: This is a little misleading, because it sounds like using `isint(<field>)` would be useful for filtering records that have an integer in any field; but in fact it would be filtering records that have an integer in the specified field. You can just copy the structure of your description for `isnum`:
> This comes in handy for conditional checks and for filtering records that have any numeric value in a field.

Then you just have to slightly change it for `isint` and `isbool`

**Comment by AndrewHess** [comment]
This looks good. I'll merge it after #110 is merged so that the link about using time variables is valid

**Review by AndrewHess — APPROVED** [review]

**Pull request merged**
